python/tk_desktop/project_menu.py

```diff
@@ -0,0 +1,150 @@
+import itertools
+
+from tank.platform.qt import QtCore, QtGui
+from sgtk.platform import get_logger
+
+log = get_logger(__name__)
+
+class ProjectMenu(object):
+    """
+    Encalsulate specific functionalities relating ot the project menu
+    """
+    def __init__(self, parent):
+        """
+
+        :param parent: DesktopWindow-->SystrayWindow-->
+        """
+        self._parent = parent
+        self._project_menu = None
+        self._pipeline_configuration_separator = None
+
+    def add(self, action):
+        self._project_menu.insertAction(self._pipeline_configuration_separator, action)
+
+    def clear(self):
+        """
+        Clears the project specific menu
+        """
+        if self._project_menu:
+            self._project_menu.clear()
+
+        self._project_menu = QtGui.QMenu(self._parent)
+
+        self._project_menu.aboutToShow.connect(self._on_project_menu_about_to_show)
+        self._project_menu.triggered.connect(self._on_project_menu_triggered)
+        self._parent.ui.actionProject_Filesystem_Folder.setVisible(True)
+        self._project_menu.addAction(self._parent.ui.actionProject_Filesystem_Folder)
+        self._parent.ui.project_menu.setMenu(self._project_menu)
+
+    def clear_actions(self):
+        """
+        Clears actions from the project menu but keeps pipeline configurations if any
+        """
+        actions = self._project_menu.actions()
+        for action in actions:
+            text = action.text()
+            print text
+            if action == self._pipeline_configuration_separator:
```

Comment:
A small comment explaining why we are breaking the loop here, e.g. `# the context's actions are listed above the separator, so we know we've removed all of them when we find it`